src/Spice86.Core/Emulator/OperatingSystem/Devices/CharacterDevice.cs

```diff
@@ -1,22 +1,17 @@
 namespace Spice86.Core.Emulator.OperatingSystem.Devices;
 
 using Spice86.Core.Emulator.OperatingSystem.Enums;
-using Spice86.Core.Emulator.OperatingSystem.Structures;
 using Spice86.Shared.Interfaces;
 
+using System;
+using System.Diagnostics.CodeAnalysis;
+using System.IO;
+
 /// <summary>
 /// Character devices are things like the console, the printer, the clock, etc.
 /// </summary>
-public class CharacterDevice : VirtualDeviceBase {
-    /// <summary>
-    /// The logging service.
-    /// </summary>
-    protected readonly ILoggerService Logger;
-
-    /// <summary>
-    /// 8-byte field with the name of the device.
-    /// </summary>
-    public string Name { get; }
```

Comment:
This and Signature could use a Range attribute.

I think it ensures at runtime that limits are enforced. Rarely used it.